File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/PublishersForm.ts
import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface PublishersForm {
    PublisherName: StringEditor;
    Contact: StringEditor;
    ContactPhone: StringEditor;
    Email: StringEditor;
}

export class PublishersForm extends PrefixedContext {
    static readonly formKey = 'BookManage.Publishers';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PublishersForm.init)  {
            PublishersForm.init = true;

            var w0 = StringEditor;

            initFormType(PublishersForm, [
                'PublisherName', w0,
                'Contact', w0,
                'ContactPhone', w0,
                'Email', w0
            ]);
        }
    }
}

File: LMIS/LMIS.Web/Modules/BookManage/Publishers/PublishersForm.cs
using Serenity.ComponentModel;
using System;

namespace LMIS.BookManage.Forms;

[FormScript("BookManage.Publishers")]
[BasedOnRow(typeof(PublishersRow), CheckNames = true)]
public class PublishersForm
{
    public string PublisherName { get; set; }
    public string Contact { get; set; }
    public string ContactPhone { get; set; }
    public string Email { get; set; }
}